### DISCUSSION TITLE
When building an archive for an iOS Cordova application, various errors are returned, but the configuration does not change

### DISCUSSION THREAD

**Discussion by alexander-mikholap** [codemagic-ci-cd/codemagic-docs#2038] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Ionic/Cordova

### Steps to reproduce

I have a problem with building a iOS application. My project is a monorepo and is built on cordova, so I use my own scripts to build the application. But every time I run the build, I may get a different build error. Sometimes it returns that it cannot find the library, but "npm install" command runs successfully, sometimes it returns that the archive failed, but the codemagic.yaml configuration doesn't change, and the result is always different. However, if I build locally, there are no such problems. I guess that I could have done something wrong, but I don't understand why it returns different errors

My codemagic.yaml:

vlaw-ios-staging:
    name: vlaw ios staging
    max_build_duration: 60
    working_directory: .
    environment:
      node: 14.20.0
      npm: 6.14.17
      ios_signing:
        provisioning_profiles:
          - LawLocker_Staging_Profile
        certificates:
          - Codemagic_Development_Certificate
      groups:
        - staging
    scripts:
      - name: Building ios app
        script: |
          brew install libpng
          echo $ENV | base64 --decode > ./.staging.env
          npm install
          npm run bootstrap:dev
          brew install openvpn
          echo 'export PATH=$PATH:/usr/local/Cellar/openvpn/2.4.8/sbin' >> ~/.zprofile
          source ~/.zprofile
          mkdir ~/decode
          echo $OPEN_VPN_CONFIG | base64 --decode > ~/decode/openvpn.ovpn
          sudo -b openvpn ~/decode/openvpn.ovpn
          npm run build:web
          npm run build:cordova:ios:release:staging:law

### Expected results

Archive that I can send to testflight

### Actual results

First error:

lerna notice cli v5.4.3
lerna info versioning independent
lerna info ci enabled
notice filter including "@silverstreet/mssql-graphql"
info filter [ '@silverstreet/mssql-graphql' ]
lerna info Executing command in 1 package: "npm run build"
/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js:139
  throw error;
  ^

Error: Cannot find module 'react-loadable/babel'
Require stack:
- /Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js
- /Users/builder/clone/node_modules/@babel/core/lib/config/files/index.js
- /Users/builder/clone/node_modules/@babel/core/lib/index.js
- /Users/builder/clone/node_modules/@babel/register/lib/worker/babel-core.js
- /Users/builder/clone/node_modules/@babel/register/lib/worker/handle-message.js
- /Users/builder/clone/node_modules/@babel/register/lib/worker-client.js
- /Users/builder/clone/node_modules/@babel/register/lib/node.js
- /Users/builder/clone/node_modules/@babel/register/lib/nodeWrapper.js
- /Users/builder/clone/node_modules/@babel/register/lib/index.js
- /Users/builder/clone/node_modules/@babel/node/lib/_babel-node.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at resolve (internal/modules/cjs/helpers.js:107:19)
    at tryRequireResolve (/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js:157:9)
    at resolveStandardizedNameForRequrie (/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js:195:19)
    at sync (/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js:224:12)
    at sync (/Users/builder/clone/node_modules/gensync/index.js:182:19)
    at /Users/builder/clone/node_modules/gensync/index.js:210:24
    at Generator.next (<anonymous>)
    at resolvePlugin (/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js:83:17)
    at resolvePlugin.next (<anonymous>) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/builder/clone/node_modules/@babel/core/lib/config/files/plugins.js',
    '/Users/builder/clone/node_modules/@babel/core/lib/config/files/index.js',
    '/Users/builder/clone/node_modules/@babel/core/lib/index.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/worker/babel-core.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/worker/handle-message.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/worker-client.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/node.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/nodeWrapper.js',
    '/Users/builder/clone/node_modules/@babel/register/lib/index.js',
    '/Users/builder/clone/node_modules/@babel/node/lib/_babel-node.js'
  ]
}

Second error: 
silverstreet/client: warning: Run script build phase 'Copy www directory' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'vlaw' from project 'vlaw')
@silverstreet/client: warning: Run script build phase 'Remove Unused Architectures' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'vlaw' from project 'vlaw')
@silverstreet/client: /Users/builder/clone/packages/wl_vlaw/dist/cordova/release/staging/platforms/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'GoogleDataTransport' from project 'Pods')
@silverstreet/client: /Users/builder/clone/packages/wl_vlaw/dist/cordova/release/staging/platforms/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 9.0, but the range of supported deployment target versions is 11.0 to 16.4.99. (in target 'GoogleAppMeasurement' from project 'Pods')
@silverstreet/client: ** ARCHIVE FAILED **
@silverstreet/client: xcodebuild: Command failed with exit code 65
@silverstreet/client: cordova build exited with code 65
@silverstreet/client: Error: cordova build command exited with code 65
@silverstreet/client:     at ChildProcess.eval (webpack-internal:///./packages/webpack_plugin_cordova/src/index.js:641:35)
@silverstreet/client:     at ChildProcess.emit (events.js:400:28)
@silverstreet/client:     at ChildProcess.emit (domain.js:475:12)
@silverstreet/client:     at maybeClose (internal/child_process.js:1088:16)
@silverstreet/client:     at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
@silverstreet/client: Process exited with code 1
@silverstreet/client: npm ERR! code ELIFECYCLE
@silverstreet/client: npm ERR! errno 1

### Build id (optional)

65004ef14b6acb6495949909

**Comment by himesh-cm** [comment]
Hi @alexandermikholap, The issue is with `provisioning profiles`.
```
error: No profiles were found: Xcode couldn't find any iOS App Development provisioning profiles 
matching the bundle id.
```
Please check under `Teams > Settings > Code signing identities` if you have uploaded the correct profile and it is valid.
If you see `red cross`, it means there is some issue with the profile.

**Reply by himesh-cm** [reply]
hi @alexandermikholap, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
